core/src/main/scala/kafka/log/Log.scala

```diff
@@ -2011,7 +2011,11 @@ object Log extends Logging {
       logDirFailureChannel,
       config.messageFormatVersion.recordVersion,
       s"[Log partition=$topicPartition, dir=${dir.getParent}] ")
-    val producerStateManager = new ProducerStateManager(topicPartition, dir, maxProducerIdExpirationMs)
+    val producerStateManager = new ProducerStateManager(
+      topicPartition,
+      dir,
+      maxProducerIdExpirationMs,
+      time)
```

Comment:
Nit: is this really easier to read than just adding the `time` parameter in the same line?

Reply:
Both are equivalent for me. I found that the line was getting too long with the addition of `time` so I broke it down that way to stay inline with `LogLoader.load` below. Would you prefer to keep it on one line instead?

Reply:
I prefer one line in this case, but I'm fine either way. :)

Reply:
Reverted to using one line.